### ISSUE TITLE
Lambda.getPolicy ignores Qualifier parameter, so returns incorrect policy results

### ISSUE THREAD

**Issue by e-gineer** [aws/aws-sdk-js#807]
According to the [Lambda docs for GetPolicy](http://docs.aws.amazon.com/lambda/latest/dg/API_GetPolicy.html) and the [AWS JS SDK Lamdba.getPolicy docs](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Lambda.html#getPolicy-property) the Qualifier property should work with Lambda.getPolicy.

Unfortunately, it seems to be ignored and calls to getPolicy always return the policy associated with the base function definition (not the qualified request).

To reproduce:

1. Create a Lambda function.
2. Create a new version of the function (now there is version 1, version $LATEST and the plain function).
3. Associate a policy (e.g. by adding a source like S3 in the console) with the function (not a specific version of the function).
4. Associate a policy (e.g. by adding a source like S3 in the console) specifically with version 1.
5. Run the getPolicy code for the function (e.g. accountId/functionName).
6. The correct policy is returned.
7. Run the getPolicy code for the function with a qualifier (e.g. accountId/functionName and 1).
8. The INCORRECT policy is returned (the function policy, same as 6 above) rather than the specific and different policy for version 1.

I believe this is because the API JSON definition is wrong and doesn't support the Qualifier parameter.

Changing the GetPolicy.input.Qualifier definition from:

```
345           "Qualifier": {}
```

to:

```
345           "Qualifier": {
346             "location": "querystring",
347             "locationName": "Qualifier"
348           }
```

appears to fix the problem.

Could you please investigate and merge a fix for this? (This is pretty bad since it effectively returns the wrong value for various cases!)

Thanks.

**Comment by chrisradek** [comment]
@e-gineer 
Thanks for finding this issue! I was able to reproduce it using the steps you provided. It seems likely this may affect other SDKs as well so the specific fix may depend on that. 

As a temporary work-around, you can include the qualifier as part of the function name.  For example, instead of setting a `Qualifier` of `1` to indicate the 1st version of the function, you could append `:1` to your `FunctionName`.

**Comment by e-gineer** [comment]
Thanks @chrisradek. The workaround is functioning well, and ironically was my preferred way to call the code in the first place. I was actually splitting out the Qualifier specifically to provide it separately <grin>.

From the Lambda API docs and SDK docs it's fairly confusing at this point about the use of the qualifier in the FunctionName. Regex says it's allowed, but docs don't mention it, etc.

From the API response and doc definition, I assume this isn't so much a temporary workaround as a model that can be used into the future even after the Qualifier specification is fixed?

**Comment by chrisradek** [comment]
@e-gineer 
Based on the Lambda API docs, it looks like including the qualifier as part of the function name is allowed. A max length for function name is also mentioned, so perhaps the `Qualifier` parameter is a way to handle long aliases.

**Comment by chrisradek** [comment]
This has been fixed with version 2.2.35 of the SDK. Now the `Qualifier` parameter is honored, and the workaround of including the qualifier as part of the function name still works as well.

**Comment by e-gineer** [comment]
Perfect - thanks!
